Add ChannelIDChangedEvent type to SlackEvent union type

### Description

Hello,

Any reason for the `ChannelIDChangedEvent` event not to be listed as a `SlackEvent`?

Thanks !

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] example code related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/bolt/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

#### Reproducible in:

package version: 3.9.0, but not in on main either at the time of this report.

#### Steps to reproduce:

1. Register an event handler like:
```
app.event('channel_id_changed', async ({ event, body }) => {
```
2. Try to access `event.old_channel_id`.
Get a Typescript error:
```
error TS2339: Property 'old_channel_id' does not exist on type 'BasicSlackEvent<"channel_id_changed">'.
```
3. Look at the type definition and notice the related event is suprisingly left out of the SlackEvent union.

#### Expected result:

No Typescript error.

#### Actual result:

See above.

--- a/src/types/events/base-events.ts
+++ b/src/types/events/base-events.ts
@@ -3,6 +3,7 @@ import { MessageEvent as AllMessageEvents } from './message-events';
 
 /**
  * All known event types in Slack's Events API
+ * Please refer to https://api.slack.com/events for more details
  *
  * This is a discriminated union. The discriminant is the `type` property.
  */

--- a/src/types/events/base-events.ts
+++ b/src/types/events/base-events.ts
@@ -16,6 +16,7 @@ export type SlackEvent =
   | ChannelCreatedEvent
   | ChannelDeletedEvent
   | ChannelHistoryChangedEvent
+  | ChannelIDChangedEvent
   | ChannelLeftEvent
   | ChannelRenameEvent
   | ChannelSharedEvent

